argus/backend/service/testrun.py

```diff
@@ -409,6 +411,25 @@ def get_github_issues(self, filter_key: str, filter_id: UUID, aggregate_by_issue
             response = [dict(issue.items()) for issue in all_issues]
         return response
 
+    def resolve_run_build_id_and_number_multiple(self, runs: list[tuple[UUID, UUID]]) -> dict[UUID, dict[str, Any]]:
+        test_ids = [r[0] for r in runs]
+        all_tests: list = []
+        for id_slice in chunk(test_ids):
+            all_tests.extend(ArgusTest.filter(id__in=id_slice).all())
+
+        tests: dict[str, ArgusTest] = {str(t.id): t for t in all_tests}
+        buckets = reduce(lambda acc, val: acc[tests[val[0]].plugin_name].append(val[1]) or acc, runs, defaultdict(list))
```

Comment:
what is a bucket?

Reply:
In this context it is a bucket of run ids keyed by plugin they use, could use a better name

Reply:
Renamed to something better.